prov/gni/src/gnix_av.c

```diff
@@ -371,15 +380,37 @@ static int map_reverse_lookup(struct gnix_fid_av *av_priv,
 {
 	GNIX_HASHTABLE_ITERATOR(av_priv->map_ht, iter);
 	struct gnix_av_addr_entry *entry;
+	fi_addr_t rx_addr;
 
 	while ((entry = _gnix_ht_iterator_next(&iter))) {
-		if (GNIX_ADDR_EQUAL(entry->gnix_addr, gnix_addr)) {
-			*fi_addr = GNIX_HASHTABLE_ITERATOR_KEY(iter);
-			return FI_SUCCESS;
+		/*
+		 * for SEP endpoint entry we may have a delta in the cdm_id
+		 * component of the address to process
+		 */
+		if ((entry->name_type & GNIX_EPN_TYPE_SEP) &&
+		    (entry->gnix_addr.device_addr == gnix_addr.device_addr)) {
+			if (gnix_addr.cdm_id - entry->gnix_addr.cdm_id <
+					entry->rx_ctx_cnt) {
+				/* we have a match */
+				int ctx_bits = 0;
+
+				rx_addr = *(fi_addr_t *)&entry->gnix_addr;
+
+				while (entry->rx_ctx_cnt >> ++ctx_bits);
```

Comment:
We insert addresses that are based on the scalable endpoint so they will always the base address of the sep.
fi_getname(&sep[i]->fid, ep_name[i], &addrlen);
fi_av_insert(av[i], ep_name[j], 1, &gni_addr[j], 0, NULL);
So, I don't think it's a concern to check if the inserted address "fits" in the rx_ctx_bits.